Dockerfile

```diff
@@ -22,6 +23,8 @@ RUN KEYFILE=/usr/share/keyrings/criu-repo-keyring.gpg; \
         crossbuild-essential-armhf \
         crossbuild-essential-ppc64el \
         crossbuild-essential-s390x \
+        binutils-riscv64-linux-gnu cpp-riscv64-linux-gnu gcc-riscv64-linux-gnu \
+        libc6-riscv64-cross libc6-dev-riscv64-cross linux-libc-dev-riscv64-cross \
```

Comment:
Perhaps put the comment before this section, and make it a "group" of things to install to make clear "these are the bits related to risc-v, by adding an "empty" line before/after e.g.;

```dockerfile
RUN KEYFILE=/usr/share/keyrings/criu-repo-keyring.gpg; \
    wget -nv $CRIU_REPO/Release.key -O- | gpg --dearmor > "$KEYFILE" \
    && echo "deb [signed-by=$KEYFILE] $CRIU_REPO/ /" > /etc/apt/sources.list.d/criu.list \
    && dpkg --add-architecture armel \
    && dpkg --add-architecture armhf \
    && dpkg --add-architecture arm64 \
    && dpkg --add-architecture ppc64el \
    && apt-get update \
    && apt-get install -y --no-install-recommends \
        build-essential \
        criu \
        crossbuild-essential-arm64 \
        crossbuild-essential-armel \
        crossbuild-essential-armhf \
        crossbuild-essential-ppc64el \
        crossbuild-essential-s390x \
        \
# no "crossbuild-essential-riscv64" package yet: https://wiki.debian.org/RISC-V
        binutils-riscv64-linux-gnu \
        cpp-riscv64-linux-gnu \
        gcc-riscv64-linux-gnu \
        libc6-dev-riscv64-cross \
        libc6-riscv64-cross \
        linux-libc-dev-riscv64-cross \
        \
        curl \
        gawk \
        gcc \
        gperf \
        iptables \
        jq \
        kmod \
        pkg-config \
        python3-minimal \
        sshfs \
        sudo \
        uidmap \
    && apt-get clean \
    && rm -rf /var/cache/apt /var/lib/apt/lists/* /etc/apt/sources.list.d/*.list
```

Reply:
Maybe [xx](https://github.com/tonistiigi/xx/) could be useful so you don't need to install all cross packages for each platform but relies on platform args in global scope. Ofc it depends if you're using buildx/BuildKit to build. See https://github.com/moby/moby/pull/43529 and https://github.com/crazy-max/moby/blob/ed158db3ad1c31300a59ccd0e70c52384d5d7308/Dockerfile.cross#L108-L123

Reply:
done.

Not sure we are allowed to use buildx/buildkit here.